frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget/GroupMappingsWidget.jsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable react/prop-types */
+import React, { useState, useEffect } from "react";
+
+import { t } from "ttag";
+import _ from "underscore";
+import AdminContentTable from "metabase/components/AdminContentTable";
+import { PermissionsApi, SettingsApi } from "metabase/services";
+import { isDefaultGroup } from "metabase/lib/groups";
+
+import Icon from "metabase/components/Icon";
+import Tooltip from "metabase/core/components/Tooltip";
+import SettingToggle from "../SettingToggle";
+import AddMappingRow from "./AddMappingRow";
+import {
+  GroupMappingsWidgetRoot as Root,
+  GroupMappingsWidgetHeader as Header,
+  GroupMappingsWidgetToggleRoot as ToggleRoot,
+  GroupMappingsWidgetAbout as About,
+  GroupMappingsWidgetAboutContentRoot as AboutContentRoot,
+  AddMappingButton,
+} from "./GroupMappingsWidget.styled";
+import MappingRow from "./MappingRow";
+
+const groupIsMappable = group => !isDefaultGroup(group);
+
+function GroupMappingsWidget({ mappingSetting, ...props }) {
+  const [showAddRow, setShowAddRow] = useState(false);
+  const [groups, setGroups] = useState(null);
+  const [mappings, setMappings] = useState({});
+  const [saveError, setSaveError] = useState(null);
+
+  useEffect(() => {
+    async function fetchData() {
+      const setting = _.findWhere(await SettingsApi.list(), {
```

Comment:
Fetching data here can be simplified with https://streamich.github.io/react-use/?path=/story/side-effects-useasync--docs